fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1035,20 +1035,30 @@ public void rebuildIndex(@Nonnull FDBRecordStore store) {
     @VisibleForTesting
     @Nonnull
     CompletableFuture<Void> buildIndexAsync(boolean markReadable) {
+        KeyValueLogMessage message = KeyValueLogMessage.build("online build index",
+                LogMessageKeys.SHOULD_MARK_READABLE, markReadable);
         if (!useSynchronizedSession) {
+            message.addKeyAndValue(LogMessageKeys.SESSION_ID, "none");
             synchronizedSessionRunner = null;
-            return handleStateAndDoBuildIndexAsync(markReadable);
+            return handleStateAndDoBuildIndexAsync(markReadable, message);
         }
         return runner
                 .runAsync(context -> openRecordStore(context).thenApply(store -> indexBuildLockSubspace(store, index)))
                 .thenCompose(lockSubspace -> runner.startSynchronizedSessionAsync(lockSubspace, leaseLengthMills))
                 .thenCompose(synchronizedRunner -> {
                     if (this.synchronizedSessionRunner != null) {
-                        throw new RecordCoreException("another synchronized session is running on the indexer");
+                        throw new RecordCoreException("another synchronized session is running on the indexer",
+                                LogMessageKeys.SESSION_ID, this.synchronizedSessionRunner.getSessionId());
                     }
                     this.synchronizedSessionRunner = synchronizedRunner;
-                    return handleStateAndDoBuildIndexAsync(markReadable)
-                            .whenComplete((vignore, e) -> {
+                    message.addKeyAndValue(LogMessageKeys.SESSION_ID, synchronizedRunner.getSessionId());
+                    return handleStateAndDoBuildIndexAsync(markReadable, message)
+                            .whenComplete((vignore, ex) -> {
+                                if (LOGGER.isWarnEnabled() && (ex != null)) {
+                                    LOGGER.warn(message.toString(), ex);
+                                } else if (LOGGER.isInfoEnabled()) {
+                                    LOGGER.info(message.toString());
```

Comment:
It might be good to include in here somewhere (other than the log level) whether the index build completed successfully or not.

Reply:
Done.